openmc/mgxs/groups.py

```diff
@@ -299,3 +303,129 @@ def merge(self, other):
         # Assign merged edges to merged groups
         merged_groups.group_edges = list(merged_edges)
         return merged_groups
+
+
+class DelayedGroups(object):
+    """A delayed groups structure used for multi-delayed-group parameters.
+
+    Parameters
+    ----------
+    groups : Iterable of Int
+        The delayed groups
+
+    Attributes
+    ----------
+    groups : Iterable of Int
+        The delayed groups
+    num_groups : int
+        The number of delayed groups
+
+    """
+
+    def __init__(self, groups=None):
+        self._groups = None
+
+        if groups is not None:
+            self.groups = groups
+
+    def __deepcopy__(self, memo):
+        existing = memo.get(id(self))
+
+        # If this is the first time we have tried to copy object, create copy
+        if existing is None:
+            clone = type(self).__new__(type(self))
+            clone._groups = copy.deepcopy(self.groups, memo)
+
+            memo[id(self)] = clone
+
+            return clone
+
+        # If this object has been copied before, return the first copy made
+        else:
+            return existing
+
+    def __eq__(self, other):
+        if not isinstance(other, DelayedGroups):
+            return False
+        elif self.num_groups != other.num_groups:
+            return False
+        elif np.allclose(self.groups, other.groups):
+            return True
+        else:
+            return False
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __hash__(self):
+        return hash(tuple(self.groups))
+
+    @property
+    def groups(self):
+        return self._groups
+
+    @property
+    def num_groups(self):
+        return len(self.groups)
+
+    @groups.setter
+    def groups(self, groups):
+        cv.check_type('groups', groups, Iterable, Integral)
+        cv.check_greater_than('number of delayed groups', len(groups), 0)
+
+        # Check that the groups are within [1, MAX_DELAYED_GROUPS]
+        for group in groups:
+            cv.check_greater_than('delayed group', group, 0)
+            cv.check_less_than('delayed group', group, MAX_DELAYED_GROUPS + 1)
```

Comment:
There's an optional `equality` argument on `check_less_than` that you can use here.